.github/workflows/dockerbuild.yml

```diff
@@ -1,16 +1,58 @@
-name: Trigger Docker Build
+name: Docker Build
 
 on:
   push:
-    branches: 
-      - 'master'
+    branches:
+      - master
+  pull_request:
+  release:
+    types:
+      - created
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-       - name: Repository Dispatch
-         uses: peter-evans/repository-dispatch@v1
-         with:
-           token: ${{ secrets.DOCKER_REPO_TOKEN }}
-           repository: robertsLando/Zwave2Mqtt-docker
-           event-type: build
+      - uses: actions/checkout@v2
+
+      - name: Just build on PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: docker build .
+
+      - name: Login to dockerhub
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+      - uses: crazy-max/ghaction-docker-buildx@v1
+        if: ${{ github.event_name != 'pull_request' }}
+
+      - name: Dev
+        if: ${{ github.ref == 'git/refs/heads/master'}}
+        run: |
+          docker buildx build \
+          --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
+          -t ${GITHUB_REPOSITORY}:latest \
+          --push \
+          .
+
+      - name: Latest
+        if: ${{ github.event_name == 'release' }}
+        run: |
+          docker buildx build \
+          --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
+          -t ${GITHUB_REPOSITORY}:latest \
```

Comment:
You can use multiple `-t` options and merge this two steps. But I would prefer to merge all buildx xommands in a single step and set the tags like I explained in my comment [here](https://github.com/OpenZWave/Zwave2Mqtt/pull/473#issuecomment-627773095)

Reply:
ah, didn't realise that worked and was what you meant, ok cool!